packages/connect/src/connector/smoldot-light.test.ts

```diff
@@ -32,19 +32,24 @@ class SdJsonRpcDisabledError extends Error {
 
 const mockSmoldotLightFactory = () => {
   const start = (options: ClientOptions) => {
-    const addChain = (addChainOptions: AddChainOptions) => {
+    const addChain = (
+      addChainOptions: AddChainOptions,
+      isClientTerminated: () => boolean,
```

Comment:
We pass `isClientTerminated()` to `addChain`?! Why is it not part of the mock to detect when `terminate` has been called?

Reply:
It *is* part of the mock to detect whether `terminate` has been called. This is precisely the point of passing the callback to the mocked chain, so that it can "ask" the mocked client if it has been terminated. It's a pretty common inversion of control technique... I'm sure that there must be a 1000 other ways of accomplishing the same thing, but this does its job, as demoed by the failing tests that it triggered... If you have a better suggestion, though, I'm all ears.